sound/soc/soc-pcm.c

```diff
@@ -2785,25 +2794,44 @@ static int soc_dpcm_fe_runtime_update(struct snd_soc_pcm_runtime *fe, int new)
 /* Called by DAPM mixer/mux changes to update audio routing between PCMs and
  * any DAI links.
  */
-int soc_dpcm_runtime_update(struct snd_soc_card *card)
+int soc_dpcm_runtime_update(struct snd_soc_card *card,
+			    enum snd_soc_update_mode mode)
 {
 	struct snd_soc_pcm_runtime *fe;
 	int ret = 0;
 
 	mutex_lock_nested(&card->mutex, SND_SOC_CARD_CLASS_RUNTIME);
+
 	/* shutdown all old paths first */
-	for_each_card_rtds(card, fe) {
-		ret = soc_dpcm_fe_runtime_update(fe, 0);
-		if (ret)
-			goto out;
-	}
+	if (mode != SND_SOC_UPDATE_STARTUP)
+		/*
+		 * This is entered if mode == RUNTIME or SHUTDOWN. In both cases
+		 * we have to call soc_dpcm_fe_runtime_update() but only in the
+		 * SHUTDOWN case we have to set the "force" (last) parameter to
+		 * "true."
+		 * The special case of SND_SOC_UPDATE_SHUTDOWN is needed for
+		 * pipelines, partially owned by virtual machines. It is used to
+		 * signal when such a pipeline shuts down.
+		 */
+		for_each_card_rtds(card, fe) {
+			ret = soc_dpcm_fe_runtime_update(fe, false,
+					mode == SND_SOC_UPDATE_SHUTDOWN);
```

Comment:
this is easier to follow if we use a variable for prune_mode.

Reply:
@lgirdwood I agree, that using constants in function parameters like `do_stuff(1, 2, true)` isn't very self-explanatory, but adding variables for a sole purpose of documentation like
```
width = 1;
hight = 2;
border = true;
do_stuff(width, height, border);
```
looks a bit exaggerated to me... Let me add comments there instead.